implement prompt:none flow

rjw57 opened the pull request:
Implement prompt:none flow to match that proposed in uisautomation/hydra-consent-app#5. This commit is intended to keep the mock consent app and real consent app in lock step with regards to features.

Update the app to cache the current subject in the flask session and automatically grant requests if there is a user in the session. This approximates the existing Raven behaviour.

Teach the app about the "prompt:none" scope which signals that the request should be immediately rejected if there is not a user in the session.

This PR is required by uisautomation/iar-frontend#153.

Closes #1

rjw57 committed:
implement prompt:none flow

Implement prompt:none flow to match that proposed in
uisautomation/hydra-consent-app#5. This commit is intended to keep the
mock consent app and real consent app in lock step with regards to
features.

Update the app to cache the current subject in the flask session and
automatically grant requests if there is a user in the session. This
approximates the existing Raven behaviour.

Teach the app about the "prompt:none" scope which signals that the
request should be immediately rejected if there is not a user in the
session.

rjw57 committed:
add /logout endpoint

Add /logout endpoint which can be used to explicitly log the current
user out of the current session.

msb reviewed (changes requested):
maybe worth adding a comment to the consent_post/consent_get handlers

rjw57 committed:
app: provide an overview for the consent_{get,post} logic

The manner in which the constent requests are accepted/rejected is not
terribly obvious. Add explanatory comments to consent_{get,post} giving
an overview of how the process works.

rjw57 commented:
> maybe worth adding a comment to the consent_post/consent_get handlers

Done in 25297d5
